src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs

```diff
@@ -158,8 +164,9 @@ public UnitDescription ExtractUnitDescriptions(string description, IReadOnlyList
 
       protected string ValidateUnit(string unit, IReadOnlyList<IDimension> supportedDimensions)
       {
-         var dimensionForUnit = supportedDimensions.FirstOrDefault(x => x.SupportsUnit(unit, ignoreCase: true));
-         if (dimensionForUnit == null)
+         var dimensionForUnit = _dimensionFactory.DimensionForUnit(unit);
+
+         if (dimensionForUnit == null || !supportedDimensions.Contains(dimensionForUnit))
```

Comment:
@msevestre similar but I would say not exactly the same. The one is setting the dimension, and the other one the unit and one occurence gets the supported dimensions as a list while the other one from the column info. Since this is a 3-liner I would personally think it would be more reasonable to leave this as is (with the new refactored functions of course).